webdriver/tests/element_click/element_click.py

```diff
@@ -0,0 +1,111 @@
+
+def test_element_not_interactable(session, transform):
+    # 14.1 Step 5
+    session.url = inline("<div style=\"width: 500px; height: 100px; background-color: green; transform: " + transform + """;">
+                            <a href=#>aaaa</a>
+                        </div>""")
+    element = session.find.css("a", all=False)
+    response = click(session, element)
+    assert_error(response, "element not interactable")
+
+
+@pytest.mark.parametrize("element", ["""<input type=button value=Roger style="position: absolute; left: 10px; top: 10px">
+                                    <div style="position: absolute; height: 100px; width: 100px; background: rgba(255,0,0,.5); left: 10px; top: 5px"></div>""",
+                                    "<input type=button value=Roger style=\"pointer-events: none\">",
+                                    """<div style="position: absolute; height: 50px; width: 100px; background: rgba(255,0,0,.5); left: 10px; top: 50px; overflow: hidden">
+                                        ABCDEFGHIJKLMNOPQRSTUVWXYZ
+                                        <input type=text value=Federer style="position: absolute; top: 50px; left: 10px;">
+                                    </div>"""])
+
+def test_element_click_intercepted(session, element):
+    # 14.1 Step 6
+    # Test intercepted click
+    # Test no pointer events
+    # Test overflow hidden
+    session.url = inline(element)
+
+    element = session.find.css("input", all=False)
+    response = click(session, element)
+    assert_error(response, "element not interactable")
+
+
+def test_match_option_element(session):
+    # 14.1 Step 7
+    session.url = inline("""
+                        <select id=sel style="position: absolute; left: 10px; top: 50px">
+                            <option id=op1></option>
+                            <option id=op2>Roger</option>
+                            <option id=op3>Federer</option>
+                        </select>
+                        <script>
+                        document.getElementById("sel").addEventListener("focus", checkFocus => {window.focusCheck = true;});
+                        document.getElementById("sel").addEventListener("mouseover", checkMouseOver => {window.mouseOverCheck = true;});
+                        document.getElementById("sel").addEventListener("mousedown", checkMouseDown => {window.mouseDownCheck = true;});
+                        document.getElementById("sel").addEventListener("mouseup", checkMouseUp => {window.mouseUpCheck = true;});
+                        document.getElementById("sel").addEventListener("click", checkMouseClick => {window.mouseClickCheck = true;});
+                        document.getElementById("sel").addEventListener("input", checkInputEvent => {window.inputEventCheck = true;});
+                        document.getElementById("sel").addEventListener("change", checkChangeEvent => {window.changeEventCheck = true;});
```

Comment:
Can you use the same approach used in https://github.com/w3c/web-platform-tests/blob/master/webdriver/tests/interaction/element_clear.py#L13?

Reply:
fixed